### PR TITLE
[REF] Extract & share code to determine revenue recognition date.

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code extraction

Before
----------------------------------------
Code not shared, less efficient method used in one place

After
----------------------------------------
Code shared

Technical Details
----------------------------------------
We have moved the code to build contributionParams back to this form so we can start consolidating it within the form, not with an unrelated form

This also starts the process of consolidating building the 'contribution params' array

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -1884,7 +1882,6 @@ protected function preparePaidEventProcessing($params): array {
       }
 
       $params['fee_level'] = $params['amount_level'];
-      $contributionParams['total_amount'] = $params['amount'];
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/Participant.php]: moved up higher

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Event/Form/Participant.php [hunk 2]
```diff
@@ -1829,18 +1829,15 @@ public function buildEventFeeForm($form) {
...
     $lineItem = [];
     $additionalParticipantDetails = [];
-    if (Civi::settings()->get('deferred_revenue_enabled')) {
-      $eventStartDate = $this->getEventValue('start_date');
-      if (strtotime($eventStartDate) > strtotime(date('Ymt'))) {
-        $contributionParams['revenue_recognition_date'] = date('Ymd', strtotime($eventStartDate));
-      }
-    }
-    if ($this->isPaymentOnExistingContribution()) {
-      $contributionParams['total_amount'] = $this->getParticipantValue('fee_amount');
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/Participant.php]: moved up higher

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
@eileenmcnaughton there is a style issue here 
```
===========================[ Identify Files ]===========================
PHP Files:
 * CRM/Event/Form/Participant.php
===============================[ php -l ]===============================
===============================[ phpcs  ]===============================


FILE: ...est/web/sites/all/modules/civicrm/CRM/Event/Form/Participant.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 1834 | ERROR | [x] Expected 1 space after ":"; 2 found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 502ms; Memory: 26MB

Found PHP errors
```

**Comment by seamuslee001** [comment]
This looks like a good straight forward extraction adding merge on pass

**Comment by seamuslee001** [comment]
Test fails look like they relate @eileenmcnaughton

**Comment by yashodha** [comment]
@eileenmcnaughton  The test failure isn't related, right?

**Comment by eileenmcnaughton** [comment]
sigh no

**Pull request merged**